quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java

```diff
@@ -0,0 +1,98 @@
+package com.upgrad.quora.service.entity;
+
+import org.hibernate.annotations.OnDelete;
+import org.hibernate.annotations.OnDeleteAction;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.time.ZonedDateTime;
+
+
+@Entity
+@Table(name = "answer")
+@NamedQueries({
+        @NamedQuery(name = "deleteAnswerByAnswerId", query = "DELETE FROM AnswerEntity c WHERE c.uuid = :uuid"),
+        @NamedQuery(name = "getAnswerDetailsByAnswerId", query = "SELECT U FROM AnswerEntity U WHERE U.uuid = :uuid")
+
+})
+
+public class AnswerEntity {
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private int id;
+
+    @Column(name = "uuid")
+    @NotNull
+    @Size(max = 200)
+    private String uuid;
+
+    @Column(name = "ans")
+    @NotNull
+    @Size(max = 255 )
+    private String answer;
+
+    @Column(name = "date")
+    @NotNull
+    private ZonedDateTime date;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "user_id")
+    @OnDelete(action = OnDeleteAction.CASCADE)
+    private UserEntity userId;
+
+    @ManyToOne(fetch = FetchType.LAZY)
```

Comment:
Why is it LAZY

Reply:
to defer initialization of an object as long as it's possible

Reply:
Thanks for explaining